From db388ad2519ac7ee4decef045521931e097f0447 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 4 Jul 2024 17:52:35 +0200
Subject: [PATCH 1000/1695] drivers: stm32_uart: remove unused pinctrl sleep

Remove parsing of DT to retrieve the pinctrl "sleep".
And remove attribute pinctrl_sleep from stm32_uart_pdata.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 2f09f905b8a59cad5f8bc19ec68c1e55d6e83b52 drivers: stm32_uart: remove unused pinctrl sleep]
Change-Id: I2c12c35972d737207bf17c5d8acf3229c7685106
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391576
Domain-Review: Alain VOLMAT <alain.volmat@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/stm32_uart.c         | 3 ---
 core/include/drivers/stm32_uart.h | 1 -
 2 files changed, 4 deletions(-)

diff --git a/core/drivers/stm32_uart.c b/core/drivers/stm32_uart.c
index c90abc1ef..5905bc7da 100644
--- a/core/drivers/stm32_uart.c
+++ b/core/drivers/stm32_uart.c
@@ -196,9 +196,6 @@ static TEE_Result parse_dt(void *fdt, int node, struct stm32_uart_pdata *pd)
 		return res;
 
 	res = pinctrl_get_state_by_name(fdt, node, "default", &pd->pinctrl);
-	if (res)
-		return res;
-	res = pinctrl_get_state_by_name(fdt, node, "sleep", &pd->pinctrl_sleep);
 	if (res)
 		return res;
 
diff --git a/core/include/drivers/stm32_uart.h b/core/include/drivers/stm32_uart.h
index 0b1783c53..a4104b815 100644
--- a/core/include/drivers/stm32_uart.h
+++ b/core/include/drivers/stm32_uart.h
@@ -19,7 +19,6 @@ struct stm32_uart_pdata {
 	bool secure;
 	struct clk *clock;
 	struct pinctrl_state *pinctrl;
-	struct pinctrl_state *pinctrl_sleep;
 };
 
 /*
-- 
2.47.2

